File: tests/test_mcp_subagent_contract.py
#!/usr/bin/env python3
from pathlib import Path

ROOT = Path(__file__).resolve().parent.parent
source = ROOT / "src/templates/skills/konoha/references/mcp-tools-block.md"
deployed = ROOT / ".agents/skills/konoha/references/mcp-tools-block.md"

if not source.exists() or not deployed.exists():
    raise SystemExit("canonical and deployed MCP tool blocks must both exist")
if source.read_bytes() != deployed.read_bytes():
    raise SystemExit("canonical and deployed MCP tool blocks differ")

content = source.read_text(encoding="utf-8")
for phrase in ("Konoha is mandatory", "Semble is mandatory", "RTK is mandatory", "Resume safety", "mcp__semble__search", "mcp__konoha__find_skill"):
    if phrase not in content:
        raise SystemExit(f"MCP subagent contract missing: {phrase}")

print("Direct MCP subagent contract and source/deployed parity passed.")

File: tests/test_mcp_subagent_contract.js
#!/usr/bin/env node
'use strict';

/**
 * tests/test_mcp_subagent_contract.js — Validates canonical and deployed MCP tool blocks parity.
 */

const assert = require('assert');
const fs = require('fs');
const path = require('path');

const ROOT = path.resolve(__dirname, '..');
const source = path.join(ROOT, 'src/templates/skills/konoha/references/mcp-tools-block.md');
const deployed = path.join(ROOT, '.agents/skills/konoha/references/mcp-tools-block.md');

async function run() {
  console.log('Running test_mcp_subagent_contract tests...');

  assert.ok(fs.existsSync(source), 'canonical MCP tool block must exist');
  assert.ok(fs.existsSync(deployed), 'deployed MCP tool block must exist');

  const sourceBytes = fs.readFileSync(source);
  const deployedBytes = fs.readFileSync(deployed);
  assert.ok(sourceBytes.equals(deployedBytes), 'canonical and deployed MCP tool blocks differ');

  const content = fs.readFileSync(source, 'utf-8');
  const requiredPhrases = [
    'Konoha is mandatory',
    'Semble is mandatory',
    'RTK is mandatory',
    'Resume safety',
    'mcp__semble__search',
    'mcp__konoha__find_skill'
  ];

  for (const phrase of requiredPhrases) {
    assert.ok(content.includes(phrase), `MCP subagent contract missing: ${phrase}`);
  }

  console.log('✓ Direct MCP subagent contract and source/deployed parity passed.');
}

if (require.main === module) {
  run().catch(err => {
    console.error(err);
    process.exit(1);
  });
}

module.exports = { run };
